extensions/objects/templates/math-editor.directive.ts

```diff
@@ -0,0 +1,56 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for math editor.
+ */
+
+// Every editor directive should implement an alwaysEditable option. There
+// may be additional customization options for the editor that should be passed
+// in via initArgs.
+
+angular.module('oppia').directive('mathEditor', [
+  function() {
+    return {
+      restrict: 'E',
+      scope: {},
+      bindToController: {
+        value: '='
+      },
+      template: require('./math-editor.directive.html'),
+      controllerAs: '$ctrl',
+      controller: ['$element', function($element) {
+        var ctrl = this;
+        var guppyDivElt, guppyDivId, guppyInstance: Guppy;
+
+        ctrl.$onInit = function() {
```

Comment:
There is an issue here where if the creator presses the 'Esc' key while the osk is visible, the modal closes, but osk is still visible, with no way to close it. Can you see if there is a way to fix it (or at least have a close button on the osk)?

Reply:
Ah, the osk can be closed by clicking the osk button again. But, just for ease, I've added an event listener that dismisses the osk upon pressing the escape key.